### DISCUSSION TITLE
Help please. How to access http.Request?

### DISCUSSION THREAD

**Discussion by joa23** [danielgtaylor/huma#625] [Q&A]
I would like to use gorilla sessions. It requires access to http.Request. 
Since e.g. chiContext is not exported is there a way to get access to the http.Request in any way?
Thank you.

**Comment by artisanphil** [comment]
Got the same question. Did you find a solution @joa23?

**Reply by joa23** [reply]
I think I used a middleware that sets it in the context. HTH

**Comment by danielgtaylor** [comment]
My suggestion to anybody needing additional access is to copy the adapter code and just add whatever new public methods you need. The adapters are small, easy to copy/modify, and rarely change. It's very simple to copy into a `myadapter` package, modify the context struct to have a `Request() *http.Request` method, then instead of `humachi.New(...)` you do `myadapter.New(...)` and voila, you have direct access without any additional memory allocations or messing with the context values.

**Comment by akhayyat** [comment]
Doesn't this answer this question?

https://huma.rocks/features/middleware/#unwrapping
